enhancements/oc/cli-manager.md

```diff
@@ -0,0 +1,333 @@
+* There are two possible approaches for this new controller:
+  1. Controller should be installed by default:
+  * Pros for being installed by default:
+    * Available day one for all clusters
+    * All binaries can be served using single unified mechanism
+    * No user setup required to use in disconnected environments
+    * Replace existing, temporary download mechanism
+  * Cons:
+    * Additional controller running by default
+  2. Controller being optional:
+  * Pros for being optionally installed:
+    * Less resource utilization
+  * Cons:
+    * Another operator to manage and mirror for disconnected environments
+    * Separate mechanism for built-in and third-party binaries
+    * Missing plugin CRs for optional elements installed before controller is installed
+    * Won't be able to replace existing, temporary system used by console which is serving some binaries
+* Given above arguments it is reasonable to include the controller by default
+* As `krew` itself is a `kubectl` plugin, it can be invoked using either using `kubectl krew` or `oc krew`\
+* `krew` functionality baked into `oc` by default
+
+Existing methods of downloading binaries (i.e. the console) will not be affected by this proposal. For the initial implementation, supported plugins will create Plugin CRs. The plugins will be downloaded and installed using `krew`.
+This transition should be transparent for consumers.
+
+### User Stories
+
+#### Story 1
+
+As a user, I want a CLI manager for various CLIs and plugins available for Kubernetes/OpenShift and related services so that I can discover, install, and list them. If `odo` was made available by a cluster-admin, I could install it using:
+* `oc krew install odo`
+
+I could then interact with `odo`:
+* `oc odo --help`
+
+##### Example
+```text
+$ oc krew search odo
+NAME                 DESC                        LATEST       INSTALLED
+-----                -----                       -----        -----
+odo                  OpenShift Developer CLI     1.0          Not Installed
+```
+
+#### Story 2
+
+As owner of a CLI or plugin, I want to publish it to users of the cluster. I need to create a Plugin CR for my tool, or provide the required information about my tool
+to a cluster-admin for the creation of a CR:
+* Name
+* Short description
+* Long description
+* Caveats
+* Homepage
+* Version
+* Platform/architecture
+* The image:tag (and registry credentials if required)
+* The paths within the image where the binary for the given platform/architecture can be found
+
+### API Extensions
+
+A new CRD will be generated based on the types:
+* API: `config.openshift.io/v1`
+* Kind: `Plugin`
+
+```go
+// PluginSpec defines the desired state of Plugin.
+type PluginSpec struct {
+	// ShortDescription of the plugin.
+	// +required
+	ShortDescription string `json:"shortDescription,omitempty"`
```

Comment:
Nit: required != omitempty

Reply:
I updated the code for the POC to remove `omitempty` after I discovered that the CRD generator ignores `// +required` if `omitempty` is present. I'll update this with the latest types.